Added `tts_coqui()` and `tts_coqui_voices()`

howardbaek opened the pull request:
1. `tts_coqui_voices()`: Checks if coqui TTS executable is installed in the shell, then runs command `tts --list_models`, and shows all the `tts_models`
2. `tts_coqui()`: Uses the `coqui-ai/TTS` [engine](https://github.com/coqui-ai/TTS) to implement a free version of text-to-speech in the `text2speech` package. Follows format used by `tts_google()`. 
3. Added commented titles to each function in the package.

cansavvy reviewed (changes requested):
Great work on this! Thanks for the description in the PR and for the comments in the code! 

Two other things we'll probably want to add: 

1. Can you add a test for it so we can make sure that it works? You can follow the instructions here to do that: https://r-pkgs.org/testing-basics.html#sec-tests-mechanics-workflow
The test should probably involve calling these function in whatever their most typical calls would look like. 
2. Is the documentation for these functions also up to date? It looks like there is a pkgdown here so you'll want to make sure to manually re-run pkgdown::build_site(). Later we should add a GHA that will do this for us. I can work on that. I'll file an issue. But for now, just run it manually.

howardbaek commented on R/tts_backend.R:
```diff
@@ -216,6 +216,8 @@ tts_microsoft = function(
 tts_coqui <- function(
     text,
+    # path to tts
+    exec_path,
```
@cansavvy  While testing `tts_coqui()`, I did realize that not everyone's path to `tts` is `/opt/homebrew/Caskroom/miniforge/base/bin`, so I replaced it with an argument, `exec_path` for portability.

howardbaek commented:
@cansavvy  

Functions I have added since your last review: 

Inside `aaa_utils.R`:
- `set_coqui_path()`
- `coqui_assert()`
- `coqui_find()`
- `process_coqui_path()`
- At the bottom of the script, I added a couple variables, `coqui_paths_to_check` and `coqui_path_missing`

cansavvy commented:
Great! I will take a look at this. But feel free to build a stacked pr on this in the meantime like we discussed so you can keep moving.
